conftest.py

```diff
@@ -167,3 +169,12 @@ def project_dict() -> dict:
 def project(project_dict) -> Project:
     """Create a project."""
     return create_project(project_dict)
+
+
+@pytest.fixture(scope='function')
+def mocked_request() -> HttpRequest:
+    """Return a mocked HttpRequest object."""
+    request = HttpRequest()
+    setattr(request, 'session', 'session')
+    setattr(request, '_messages', FallbackStorage(request))
```

Comment:
What is the FallbackStorage needed for? From the changes in the code it is not clear to me, why we need it in the tests.

Reply:
For some reason the request objects in tests need an addtitional message middleware which can be solved by [this](https://code.djangoproject.com/ticket/17971).

Reply:
Maybe you could link the ticket here?